File: 1752.check-if-array-is-sorted-and-rotated.cpp
/*
 * @lc app=leetcode id=1752 lang=cpp
 *
 * [1752] Check if Array Is Sorted and Rotated
 */
#include <vector>
#include <climits>
using namespace std;
// @lc code=start
class Solution
{
public:
    bool check(vector<int> &nums)
    {
        int disc_cnt = 0;
        int prev = INT_MIN;
        for (auto n : nums)
        {
            if (n < prev)
            {
                if (++disc_cnt == 2)
                {
                    return false;
                }
            }
            if (disc_cnt == 1 && n > nums[0])
            {
                return false;
            }
            prev = n;
        }
        return true;
    }
};
// @lc code=end

File: 1752.check-if-array-is-sorted-and-rotated.py
#
# @lc app=leetcode id=1752 lang=python3
#
# [1752] Check if Array Is Sorted and Rotated
#

# @lc code=start
class Solution:
    def check(self, nums: List[int]) -> bool:
        disc_cnt = 0
        prev = 0
        for n in nums:
            if n < prev:
                disc_cnt += 1
                if disc_cnt == 2:
                    return False
            if disc_cnt == 1 and n > nums[0]:
                return False
            prev = n
        return True
# @lc code=end
